cwl/workflows/wf-main-toil.cwl

```diff
@@ -0,0 +1,319 @@
+
+# ------------ GTDB-Tk --------------
+#  gtdbtk:
+#    type: Directory?
+#    outputSource: gtdbtk/gtdbtk_folder
+
+steps:
+
+# ----------- << checkm for NCBI >> -----------
+  checkm_subwf:
+    run: sub-wf/checkm-subwf.cwl
+    when: $(Boolean(inputs.genomes_folder))
+    in:
+      genomes_folder: genomes_ncbi
+    out:
+      - checkm_csv
+
+# ----------- << unite NCBI and ENA >> -----------
+  unite_folders:
+    run: ../tools/unite_ena_ncbi/unite.cwl
+    in:
+      ena_folder: genomes_ena
+      ncbi_folder: genomes_ncbi
+      ena_csv: ena_csv
+      ncbi_csv: checkm_subwf/checkm_csv
+      outputname: { default: "genomes"}
+    out:
+      - genomes
+      - csv
+
+# ----------- << assign MGYGs >> -----------
+  assign_mgygs:
+    run: ../tools/rename_fasta/rename_fasta.cwl
+    in:
+      genomes: unite_folders/genomes
+      prefix: { default: "MGYG"}
+      start_number: min_accession_mgyg
+      max_number: max_accession_mgyg
+      output_filename: { default: "names.tsv"}
+      output_dirname: { default: "mgyg_genomes" }
+      csv: unite_folders/csv
+    out:
+      - naming_table
+      - renamed_genomes
+      - renamed_csv
+
+# ---------- dRep + split -----------
+
+  generate_weights:
+    when: $(inputs.flag != "skip")
+    run: ../tools/generate_weight_table/generate_extra_weight_table.cwl
+    in:
+      flag: skip_drep_step
+      input_directory: assign_mgygs/renamed_genomes
+      output: { default: "extra_weight_table.txt" }
+    out: [ file_with_weights ]
+
+  drep:
+    when: $(inputs.flag != "skip")
+    run: ../tools/drep/drep.cwl
+    in:
+      flag: skip_drep_step
+      genomes: assign_mgygs/renamed_genomes
+      drep_outfolder: { default: 'drep_outfolder' }
+      checkm_csv: assign_mgygs/renamed_csv
+      extra_weights: generate_weights/file_with_weights
+    out: [ out_folder, dereplicated_genomes ]
+
+  split_drep:
+    when: $(inputs.flag != "skip")
+    run: ../tools/drep/split_drep.cwl
+    in:
+      flag: skip_drep_step
+      genomes_folder: assign_mgygs/renamed_genomes
+      drep_folder: drep/out_folder
+      split_outfolder: { default: 'split_outfolder' }
+    out: [ split_out ]
+
+  classify_clusters:
+    when: $(inputs.flag != "skip")
+    run: ../tools/drep/classify_folders.cwl
+    in:
+      flag: skip_drep_step
+      clusters: split_drep/split_out
+    out:
+      - many_genomes
+      - one_genome
+      - mash_folder
+      - stderr
+      - stdout
+
+  classify_dereplicated:
+    when: $(inputs.flag == "skip")
+    run: ../tools/drep/classify_dereplicated.cwl
+    in:
+      flag: skip_drep_step
+      clusters: assign_mgygs/renamed_genomes
+    out:
+      - one_genome
+
+  filter_nulls:
+    run: ../utils/filter_nulls.cwl
+    in:
+      list_dirs:
+        source:
+          - classify_clusters/one_genome
+          - classify_dereplicated/one_genome
+        linkMerge: merge_flattened
+    out: [ out_dirs ]
+
+
+# ---------- annotation
+  clusters_annotation:
```

Comment:
Maybe it's worth splitting this into 2:
- many_genomes
- one_genome
it could improve the readability in subwf-process_clusters.cwl?